app/src/main/java/com/example/weatherapp/mvp/presenter/WeatherPresenter.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.weatherapp.mvp.presenter
+
+import com.example.weatherapp.mvp.WeatherContract
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.schedulers.Schedulers
+
+class WeatherPresenter(
+    private val model: WeatherContract.Model,
+    private val view: WeatherContract.View
+) : WeatherContract.Presenter {
+
+    override fun getWeather() {
+        model.getData(CITY)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { data -> view.showData(data) }
```

Comment:
add the error option. If you call this without network your app will crash

Reply:
Solved, thanks!